extensions/VisualEditor/lib/ve/lib/jquery.client/jquery.client.js

```diff
@@ -0,0 +1,299 @@
+		 * algorithm, similar to PHP's version_compare ('1.2' < '1.11').
+		 *
+		 * A browser map is in the following format:
+		 * {
+		 *   // Multiple rules with configurable operators
+		 *   'msie': [['>=', 7], ['!=', 9]],
+		 *    // Match no versions
+		 *   'iphone': false,
+		 *    // Match any version
+		 *   'android': null
+		 * }
+		 *
+		 * It can optionally be split into ltr/rtl sections:
+		 * {
+		 *   'ltr': {
+		 *     'android': null,
+		 *     'iphone': false
+		 *   },
+		 *   'rtl': {
+		 *     'android': false,
+		 *     // rules are not inherited from ltr
+		 *     'iphone': false
+		 *   }
+		 * }
+		 *
+		 * @param {Object} map Browser support map
+		 * @param {Object} [profile] A client-profile object
+		 * @param {boolean} [exactMatchOnly=false] Only return true if the browser is matched, otherwise
+		 * returns true if the browser is not found.
+		 *
+		 * @returns {boolean} The current browser is in the support map
+		 */
+		test: function ( map, profile, exactMatchOnly ) {
+			/*jshint evil: true */
+
+			var conditions, dir, i, op, val, j, pieceVersion, pieceVal, compare;
+			profile = $.isPlainObject( profile ) ? profile : $.client.profile();
+			if ( map.ltr && map.rtl ) {
+				dir = $( 'body' ).is( '.rtl' ) ? 'rtl' : 'ltr';
+				map = map[dir];
+			}
+			// Check over each browser condition to determine if we are running in a compatible client
+			if ( typeof map !== 'object' || map[profile.name] === undefined ) {
+				// Not found, return true if exactMatchOnly not set, false otherwise
+				return !exactMatchOnly;
+			}
+			conditions = map[profile.name];
+			if ( conditions === false ) {
+				// Match no versions
+				return false;
+			}
+			if ( conditions === null ) {
+				// Match all versions
+				return true;
+			}
+			for ( i = 0; i < conditions.length; i++ ) {
+				op = conditions[i][0];
+				val = conditions[i][1];
+				if ( typeof val === 'string' ) {
+					// Perform a component-wise comparison of versions, similar to PHP's version_compare
+					// but simpler. '1.11' is larger than '1.2'.
+					pieceVersion = profile.version.toString().split( '.' );
+					pieceVal = val.split( '.' );
+					// Extend with zeroes to equal length
+					while ( pieceVersion.length < pieceVal.length ) {
+						pieceVersion.push( '0' );
+					}
+					while ( pieceVal.length < pieceVersion.length ) {
+						pieceVal.push( '0' );
+					}
+					// Compare components
+					compare = 0;
+					for ( j = 0; j < pieceVersion.length; j++ ) {
+						if ( Number( pieceVersion[j] ) < Number( pieceVal[j] ) ) {
+							compare = -1;
+							break;
+						} else if ( Number( pieceVersion[j] ) > Number( pieceVal[j] ) ) {
+							compare = 1;
+							break;
+						}
+					}
+					// compare will be -1, 0 or 1, depending on comparison result
+					if ( !( eval( '' + compare + op + '0' ) ) ) {
+						return false;
+					}
+				} else if ( typeof val === 'number' ) {
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) Merge this if statement with the nested one. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonarqube.wikia-inc.com/coding_rules#rule_key=javascript%3ACollapsibleIfStatements)